src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs

```diff
@@ -123,34 +124,49 @@ SyntaxNodeAnalysisContext ctx
 				);
 
 				return;
-			}
-
-			// Literal arguments should always be named
-			foreach( var arg in unnamedArgs ) {
-
-				// Check if the argument type is literal
-				if( arg.Syntax.Expression is LiteralExpressionSyntax ) {
-
+			}
+
+			// Generate a Dictionary of data type usages
+			Dictionary<string, int> dataTypeCount = new Dictionary<string, int>();
```

Comment:
This doesn't really handle all the cases unfortunately.

What we want to do is catch cases where arguments could have been swapped, e.g.:

```cs
interface IA {}
interface IB : IA {}
interface IC : IA {}
interface ID : IA, IB {}
// assume impls A, B, C, D

void Foo( IA x, IB y ) {}

// all of these compile. they all base a B (which impls IB) as the second arg,
// but we consider all cases for the first arg:
Foo( new A(), new B() ); // always safe (no argument swapping possible)
Foo( new B(), new B() ); // possible error
Foo( new C(), new B() ); // always safe
Foo( new D(), new B() ); // possible error
```

There is a way to ask the semantic model if a particular conversion exists.

Reply:
From DM: this is roughly what we want to be use, I think: [SemanticModel.ClassifyConversion](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.csharp.csharpextensions.classifyconversion?view=roslyn-dotnet#Microsoft_CodeAnalysis_CSharp_CSharpExtensions_ClassifyConversion_Microsoft_CodeAnalysis_SemanticModel_Microsoft_CodeAnalysis_CSharp_Syntax_ExpressionSyntax_Microsoft_CodeAnalysis_ITypeSymbol_System_Boolean_).

Reply:
This one was interesting, but it should be good now.